pkg/controllers/cassandra/cluster_control_test.go

```diff
@@ -49,7 +54,55 @@ func CassandraClusterStatusSummary(c *v1alpha1.CassandraCluster) string {
 func TestNextAction(t *testing.T) {
 	f := func(c *v1alpha1.CassandraCluster) bool {
 		t.Log(CassandraClusterSummary(c))
-		a := cassandra.NextAction(c)
+
+		fixture := &framework.StateFixture{
+			T: t,
+			KubeObjects: []runtime.Object{
+				generate.StatefulSet(
+					generate.StatefulSetConfig{
+						Name:      "cass-cluster1-np0",
+						Namespace: "ns1",
+						Replicas:  ptr.Int32(8),
+					},
+				),
+				generate.StatefulSet(
+					generate.StatefulSetConfig{
+						Name:      "cass-cluster1-np1",
+						Namespace: "ns1",
+						Replicas:  ptr.Int32(8),
+					},
+				),
+				generate.StatefulSet(
+					generate.StatefulSetConfig{
+						Name:      "cass-cluster1-np2",
+						Namespace: "ns1",
+						Replicas:  ptr.Int32(8),
+					},
+				),
+				generate.StatefulSet(
+					generate.StatefulSetConfig{
+						Name:      "cass-cluster1-np3",
+						Namespace: "ns1",
+						Replicas:  ptr.Int32(8),
+					},
+				),
+				generate.StatefulSet(
+					generate.StatefulSetConfig{
+						Name:      "cass-cluster1-np4",
+						Namespace: "ns1",
+						Replicas:  ptr.Int32(8),
+					},
+				),
+			},
+		}
+		fixture.Start()
+		defer fixture.Stop()
+		state := fixture.State()
+
+		a, err := cassandra.NextAction(c, state.StatefulSetLister)
+		if err != nil {
+			t.Errorf("error calculating next action: %v", err)
+		}
```

Comment:
And if we could keep the `NextAction` signature unchanged, we wouldn't need to add this extra stuff to the test.